File: app/Models/Comment.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Comment extends Model
{
    use HasFactory;

    protected $fillable = ['user_id', 'saved_road_id', 'comment'];

    public function road()
    {
        return $this->belongsTo(SavedRoad::class);
    }

    public function user()
    {
        return $this->belongsTo(User::class);
    }
}

File: android-native/app/src/main/java/com/scenicroutes/app/data/model/Comment.kt
package com.scenicroutes.app.data.model

data class Comment(
    val id: Long,
    val user_id: Long,
    val saved_road_id: Long,
    val comment: String,
    val created_at: String,
    val updated_at: String,
    val user: User? = null,
)

data class CommentRequest(
    val comment: String,
)
















